extras/rya.indexing.pcj/src/main/java/org/apache/rya/indexing/pcj/storage/mongo/MongoPcjDocuments.java

```diff
@@ -0,0 +1,466 @@
+        final int cardinality = result.getInteger(CARDINALITY_FIELD, 0);
+        final List<List<String>> varOrders= (List<List<String>>) result.get(VAR_ORDER_FIELD);
+        final Set<VariableOrder> varOrder = new HashSet<>();
+        for(final List<String> vars : varOrders) {
+            varOrder.add(new VariableOrder(vars));
+        }
+
+        return new PcjMetadata(sparql, cardinality, varOrder);
+    }
+
+    /**
+     * Adds binding set results to a specific PCJ.
+     * 
+     * @param pcjId
+     *            - The PCJ to add the results to.
+     * @param results
+     *            - The binding set results.
+     */
+    public void addResults(final String pcjId, final Collection<VisibilityBindingSet> results) {
+        checkNotNull(pcjId);
+        checkNotNull(results);
+
+        final List<Document> pcjDocs = new ArrayList<>();
+        for (final VisibilityBindingSet vbs : results) {
+            // each binding gets it's own doc.
+            final Document bindingDoc = new Document(PCJ_NAME, pcjId);
+            vbs.forEach(binding -> {
+                final RyaType type = RdfToRyaConversions.convertValue(binding.getValue());
+                bindingDoc.append(binding.getName(),
+                        new Document()
+                        .append(BINDING_TYPE, type.getDataType().stringValue())
+                        .append(BINDING_VALUE, type.getData())
+                        );
+            });
+            bindingDoc.append(VISIBILITIES_FIELD, vbs.getVisibility());
+            pcjDocs.add(bindingDoc);
+        }
+        pcjCollection.insertMany(pcjDocs);
+
+        // update cardinality in the metadata doc.
+        final int appendCardinality = pcjDocs.size();
+        final Bson query = new Document(PCJ_ID, makeMetadataID(pcjId));
+        final Bson update = new Document("$inc", new Document(CARDINALITY_FIELD, appendCardinality));
+        pcjCollection.updateOne(query, update);
+    }
+
+    /**
+     * Purges all results from the PCJ document with the provided name.
+     * 
+     * @param pcjId
+     *            - The name of the PCJ document to purge. (not null)
+     */
+    public void purgePcjs(final String pcjId) {
+        requireNonNull(pcjId);
+
+        // remove every doc for the pcj, except the metadata
+        final Bson filter = new Document(PCJ_NAME, pcjId);
+        pcjCollection.deleteMany(filter);
+
+        // reset cardinality
+        final Bson query = new Document(PCJ_ID, makeMetadataID(pcjId));
+        final Bson update = new Document("$set", new Document(CARDINALITY_FIELD, 0));
+        pcjCollection.updateOne(query, update);
+    }
+
+    /**
+     * Scan Rya for results that solve the PCJ's query and store them in the PCJ
+     * document.
+     * <p>
+     * This method assumes the PCJ document has already been created.
+     *
+     * @param pcjId
+     *            - The name of the PCJ table that will receive the results. (not
```

Comment:
I think everywhere this class mentions "PCJ table" that it is copy paste from the Accumulo implementation. It doesn't describe how it works in Mongo DB. This whole class could use a second pass to ensure it's documented with the Mongo implementation in mind.